src/index.edge.ts

```diff
@@ -1,4 +1,106 @@
+import "performance-polyfill";
+import {
+  BasicTracerProvider,
+  ReadableSpan,
+  SDKRegistrationConfig,
+  SimpleSpanProcessor,
+  TracerConfig,
+} from "@opentelemetry/sdk-trace-base";
+import { AsyncLocalStorageContextManager } from "@opentelemetry/context-async-hooks";
+import { Resource } from "@opentelemetry/resources";
+import { SemanticResourceAttributes } from "@opentelemetry/semantic-conventions";
+import {
+  OTLPExporterBase,
+  OTLPExporterError,
+  OTLPExporterConfigBase,
+} from "@opentelemetry/otlp-exporter-base";
+import { diag } from "@opentelemetry/api";
+import { createExportTraceServiceRequest } from "@opentelemetry/otlp-transformer";
+import { IExportTraceServiceRequest } from "@opentelemetry/otlp-transformer";
+import { getEnv } from "@opentelemetry/core";
+
+const DEFAULT_COLLECTOR_RESOURCE_PATH = "v1/traces";
+const DEFAULT_COLLECTOR_URL = `http://localhost:4318/${DEFAULT_COLLECTOR_RESOURCE_PATH}`;
+
+class EdgeTraceProvider extends BasicTracerProvider {
+  constructor(config: TracerConfig = {}) {
+    super(config);
+  }
+
+  register(config: SDKRegistrationConfig = {}) {
+    if (config.contextManager === undefined) {
+      config.contextManager = new AsyncLocalStorageContextManager();
+      config.contextManager.enable();
+    }
+
+    super.register(config);
+  }
+}
+
+abstract class OTLPExporterEdgeBase<
+  ExportItem,
+  ServiceRequest
+> extends OTLPExporterBase<OTLPExporterConfigBase, ExportItem, ServiceRequest> {
+  onShutdown() {}
+  onInit(_config: OTLPExporterConfigBase) {}
+  send(
+    items: ExportItem[],
+    onSuccess: () => void,
+    onError: (error: OTLPExporterError) => void
+  ): void {
+    if (this._shutdownOnce.isCalled) {
+      diag.debug("Shutdown already started. Cannot send objects");
+      return;
+    }
+    const serviceRequest = this.convert(items);
+
+    const promise = fetch(this.url, {
+      method: "POST",
+      body: JSON.stringify(serviceRequest),
+      headers: {
+        "Content-Type": "application/json",
```

Comment:
without additional headers from `OTEL_EXPORTER_OTLP_HEADERS` this won't do much for folks.